lib/graphql/language/nodes.rb

```diff
@@ -25,14 +25,35 @@ def initialize_node(options={})
           raise NotImplementedError
         end
 
+        def eql?(other)
+          other.is_a?(self.class) &&
+            other.scalars.eql?(self.scalars) &&
+            other.children.eql?(self.children)
+        end
+
         # @return [GraphQL::Language::Nodes::AbstractNode] all nodes in the tree below this one
         def children
           self.class.child_attributes
             .map { |attr_name| public_send(attr_name) }
             .flatten
         end
 
+        def scalars
+          self.class.scalar_attributes
+            .map { |attr_name| public_send(attr_name) }
+        end
+
         class << self
+          def inherited(subclass)
+            subclass.scalar_attributes(*@scalar_attributes)
+            subclass.child_attributes(*@child_attributes)
+          end
```

Comment:
:+1: for `scalar_attributes` 

For `child_attributes`,  I didn't include it because I didn't need it. Is there any place where that code is (or could be) used? 

If you want to include it for consistency, that's ok with me, but I want to make sure it's tested somehow, either by use "in the wild" (ie, some of the node classes inherit child attributes) or in a unit test with a dummy subclass.  What do you think?